File: gitdot-api-ts/src/endpoint/runner/create-runner-token.ts
import { z } from "zod";

import { RunnerTokenResource } from "../../resource";

export const CreateRunnerTokenRequest = z.object({});
export type CreateRunnerTokenRequest = z.infer<typeof CreateRunnerTokenRequest>;

export const CreateRunnerToken = {
  path: "/ci/runner/{owner}/{name}/token",
  method: "POST",
  request: CreateRunnerTokenRequest,
  response: RunnerTokenResource,
} as const;
export type CreateRunnerToken = typeof CreateRunnerToken;

File: gitdot-api/src/endpoint/runner/create_runner_token.rs
use crate::{endpoint::Endpoint, resource::runner::RunnerTokenResource};

pub struct CreateRunnerToken;

impl Endpoint for CreateRunnerToken {
    const PATH: &'static str = "/ci/runner/{owner}/{name}/token";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateRunnerTokenRequest;
    type Response = CreateRunnerTokenResponse;
}

pub type CreateRunnerTokenRequest = ();
pub type CreateRunnerTokenResponse = RunnerTokenResource;
